### PR TITLE
Dockerize the Django-React-Library Application

### PR DESCRIPTION
**Pull request by BolajiOlajide**
#### What Does This PR Do?
It sets up docker environment for development purposes.

#### Description Of Task To Be Completed
Add `Dockerfile` and `docker-compose.yml` files to detail docker operations.

#### Any Background Context You Want To Provide?
N/A

#### How can this be manually tested?
- Clone the repo
- run `docker-compose up` to start up the application.

### CODE REVIEW

**Comment by andela-sjames** [comment]
@BolajiOlajide should I be expecting the automation script later?

**Comment by BolajiOlajide** [comment]
Yes, @andela-sjames. I just raised the PR so I don't have to do it later. I'll add `WIP` to the title to indicate that it's a work in progress.

**Comment by BolajiOlajide** [comment]
Hey @andela-sjames, 
Can you check out the automation script and see what needs to be changed? Thanks

File: reactlibapp/start_app.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+function collect_static() {
```
Comment (andela-sjames) [inline reactlibapp/start_app.sh]: The function is defined but not called anywhere? Also, how do we tell the `./start_app` script to execute docker.

File: reactlibapp/start_app.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    # navigate into the static folder to install static files
+    cd static
+
```
Comment (andela-sjames) [inline reactlibapp/start_app.sh]: Should we check where we are at first before locating the static folder. Once in the static folder checks should be made to determine if node_modules and bower_components exist. If no then `Bower` and `npm` install should follow before `webpack` command is executed to generate the dist/ folder. If they exist but no `dist/` folder then just webpack should suffice. On the other hand, if all these conditions have been met, then go ahead to makemigrations/migrate and runserver...

File: reactlibapp/start_app.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    cd bower_components
+
+    if [ -d "jquery" ]; then
+        # run bower install to install static files
+        bower install
+    fi
+}
+
+function export_env() {
```
Comment (andela-sjames) [inline reactlibapp/start_app.sh]: Don't forget this file is only for development and should be taken off GitHub when ready.

Comment (BolajiOlajide) [inline reactlibapp/start_app.sh]: Corrected.

File: reactlibapp/start_app.sh (hunk 2 above)
Comment (BolajiOlajide) [inline reactlibapp/start_app.sh]: Done!

File: reactlibapp/start_app.sh (hunk 1 above)
Comment (BolajiOlajide) [inline reactlibapp/start_app.sh]: Done

File: reactlibapp/start_app.sh (hunk 2 above)
Comment (andela-sjames) [inline reactlibapp/start_app.sh]: We should have a `./start_app local` functionality in addition to a `./start_app docker` functionality. @BolajiOlajide

**Comment by andela-sjames** [comment]
This should be made across `develop` branch now. Thanks.

**Pull request closed without merging**